app/src/editor/shared/components/Modal.jsx

```diff
@@ -162,13 +166,17 @@ export default class ModalWithOverlay extends React.Component {
     render() {
         const { children, modalId, overlayClassName } = this.props
         return (
-            <Modal modalId={modalId}>
-                {({ value, api }) => {
+            <ModalContainer modalId={modalId}>
```

Comment:
Ok so the idea with the `Modal` component vs `ModalWithOverlay` component is that the `Modal` just implemented simplest thing that would work for "modal" behaviour, whereas the `ModalWithOverlay` wraps this with the overlay/esc to close behaviours.

I think the design I was trying to set up was to allow the basic modal implementation to still be useful in contexts where you don't need background overlay/esc to close, however in practice this `Modal` export isn't actually currently used by anything, and with this change it's no longer even used internally so… `Modal` really could just be removed.

Reply:
Ok, you mean that `ModalWithOverlay` should become `Modal` and use it only? Using `Modal` inside `ModalWithOverlay` caused the overlay to appear on top of the contents, that's why it's duplicated here.

Reply:
Hmm, does the focusing fail because the input ref is not set?